Mathlib/FieldTheory/RatFunc.lean

```diff
@@ -197,7 +197,7 @@ theorem mk_eq_div' (p q : K[X]) :
     RatFunc.mk p q = ofFractionRing (algebraMap _ _ p / algebraMap _ _ q) := by rw [RatFunc.mk]
 #align ratfunc.mk_eq_div' RatFunc.mk_eq_div'
 
-theorem mk_zero (p : K[X]) : RatFunc.mk p 0 = ofFractionRing 0 := by
+theorem mk_zero (p : K[X]) : RatFunc.mk p 0 = ofFractionRing (0 : FractionRing K[X]) := by
```

Comment:
This one times out without the type ascription, because Lean 4 will happily go off and synthesize a ridiculous instance on the type with metavars `(?K[X])⁰` that it then takes several seconds to (correctly) fail unifying with the actual instance before it can backtrack. Apparently before tweaking the priorities, the wrong instance was more obviously wrong.

This type ascription brings the elaboration time down from about half a second (on master) / 20 seconds (on the previous commit) to about a quarter of a second (on both branches).